public/app/features/explore/ExploreQueryInspector.tsx

```diff
@@ -53,14 +55,24 @@ export function ExploreQueryInspector(props: Props) {
     ),
   };
 
-  const queryInspectorTab: TabConfig = {
-    label: 'Query Inspector',
-    value: 'query_inspector',
+  const errorTab: TabConfig = {
+    label: 'Error',
+    value: 'error',
+    icon: 'exclamation-triangle',
+    content: <InspectErrorTab error={error} />,
+  };
+
+  const queryTab: TabConfig = {
+    label: 'Query',
+    value: 'query',
     icon: 'info-circle',
     content: <QueryInspector data={dataFrames} onRefreshQuery={() => props.runQueries(props.exploreId)} />,
   };
 
-  const tabs = [statsTab, queryInspectorTab, jsonTab, dataTab];
+  const tabs = [statsTab, queryTab, jsonTab, dataTab];
+  if (error) {
+    tabs.push(errorTab);
```

Comment:
i recommend moving the whole `const errorTab : TabConfig = {...}` code here, inside the `if (error)`, so that the `errorTab` object is only created when there is an error.

the code works in the current form too, i just think it would be easier to understand with the change.

Reply:
Good point! I totally agree. Updated in https://github.com/grafana/grafana/pull/33412/commits/62d8ee02d62d48b9561b54aa763aa6124e32d920